src/renderer/widgets/RemoveProxy/model/remove-proxy-model.ts

```diff
@@ -0,0 +1,209 @@
+
+    return (
+      accounts.find(
+        (a) => a.accountId === proxyAccount.proxiedAccountId && !walletUtils.isWatchOnly(walletsMap[a.walletId]),
+      ) || null
+    );
+  },
+  {
+    skipVoid: false,
+  },
+);
+
+const $proxiedWallet = combine(
+  {
+    wallets: walletModel.$wallets,
+    proxiedAccount: $proxiedAccount,
+  },
+  ({ wallets, proxiedAccount }) => {
+    if (!proxiedAccount) return null;
+
+    return wallets.find((w) => w.id === proxiedAccount.walletId) || null;
+  },
+  {
+    skipVoid: false,
+  },
+);
+
+const $signatories = combine(
+  {
+    wallets: walletModel.$wallets,
+    accounts: walletModel.$accounts,
+    proxiedAccount: $proxiedAccount,
+    chain: $chain,
+  },
+  ({ wallets, accounts, proxiedAccount, chain }) => {
+    if (!chain || !proxiedAccount) return null;
+
+    return accountUtils.isMultisigAccount(proxiedAccount)
+      ? getSignatoryAccounts(accounts, wallets, [], proxiedAccount.signatories, chain.chainId)
+      : null;
+  },
+  {
+    skipVoid: false,
+  },
+);
+
+const signatorySelected = createEvent<Account>();
+const activeStepChanged = createEvent<Step>();
+
+type TransactionSignedProps = {
+  signature: HexString;
+  unsignedTx: UnsignedTransaction;
+};
+const transactionSigned = createEvent<TransactionSignedProps>();
+const proxyRemoved = createEvent<ApiPromise>();
+type ProxyGroupUpdatedProps = {
+  proxyGroup: ProxyGroup;
+  shouldDelete: boolean;
+};
+const proxyGroupUpdated = createEvent<ProxyGroupUpdatedProps>();
+
+sample({
+  source: flowStarted,
+  target: spread({
+    targets: { chain: $chain, proxyAccount: $proxyAccount },
+  }),
+});
+
+sample({
+  source: transactionSigned,
+  target: spread({
+    targets: { unsignedTx: $unsignedTx, signature: $signature },
+  }),
+});
+
+sample({
+  clock: transactionSigned,
+  fn: () => Step.SUBMIT,
+  target: $activeStep,
+});
+
+sample({
+  source: signatorySelected,
+  target: $signatory,
+});
+
+sample({
+  source: signatorySelected,
+  fn: () => Step.SIGNING,
+  target: $activeStep,
+});
+
+sample({
+  source: activeStepChanged,
+  target: $activeStep,
+});
+
+sample({
+  clock: proxyRemoved,
+  source: {
+    proxyAccount: $proxyAccount,
+    proxies: proxyModel.$proxies,
+    proxyGroups: proxyModel.$proxyGroups,
+  },
+  filter: (proxyAccount) => Boolean(proxyAccount),
+  fn: ({ proxyAccount, proxies, proxyGroups }, api) => {
+    const sameGroupProxies = proxies[proxyAccount!.proxiedAccountId].filter((p) => p.chainId === proxyAccount!.chainId);
+    const proxyGroup = proxyGroups.find(
+      (p) => p.chainId === proxyAccount!.chainId && p.proxiedAccountId === proxyAccount!.proxiedAccountId,
+    );
+
+    if (sameGroupProxies.length === 1 && proxyGroup) {
+      return { proxyGroup, shouldDelete: true };
+    }
+
+    const updatedDeposit =
+      Number(proxyGroup!.totalDeposit) -
+      api.consts.proxy.proxyDepositBase.toNumber() -
+      api.consts.proxy.proxyDepositBase.toNumber();
```

Comment:
Better to use `BN` here

And maybe you should subtract `proxyDepositFactor` not `proxyDepositBase`? As I clearly understand `proxyDepositBase` used only for first (last on remove) proxy

Reply:
yeah, second one should be `proxyDepositFactor`